app/src/main/java/org/mozilla/fenix/settings/account/SignOutFragment.kt

```diff
@@ -66,9 +66,13 @@ class SignOutFragment : AppCompatDialogFragment() {
                 accountManager.logout()
             }.invokeOnCompletion {
                 runIfFragmentIsAttached {
-                    if (!findNavController().popBackStack(R.id.settingsFragment, false)) {
+                    if (this.isVisible) {
                         dismiss()
                     }
+
+                    if (!findNavController().popBackStack(R.id.settingsFragment, false))
+                        if (!findNavController().popBackStack(R.id.browserFragment, false))
+                            findNavController().popBackStack(R.id.homeFragment, false)
```

Comment:
Do we need to go through the back stack like this? Could we use `popBackStack` and rely on the nav graph to know where we need to be?

Alternatively, if we need to keep this logic, it is hard to read. Could we try to expand on it to be clear?

Something like the logic below should work the same and a one-line comment will give enough context for the next person reading the code some context.

```suggestion
                    // If we can't go to settings or the browser, then let's go home.
                    if (!findNavController().popBackStack(R.id.settingsFragment, false) || findNavController().popBackStack(R.id.browserFragment, false)) {
                            findNavController().popBackStack(R.id.homeFragment, false)
                     }
```

Reply:
>Could we use popBackStack and rely on the nav graph to know where we need to be?

You're right. 🤦  I don't know what I was thinking.
Thanks for taking a look at this 🥇